app/controllers/transition/NeedToPayTaxYesNoController.scala

```diff
@@ -86,13 +87,17 @@ class NeedToPayTaxYesNoController @Inject()(
       )
   }
 
+  private def updateTransforms(hasAnswerChanged: Boolean, needsToPayTax: Boolean)
+                              (implicit request: DataRequest[AnyContent]): Future[HttpResponse] = {
 
-  private def makeRequestIfConditionMet(condition: Boolean)(request: => Future[HttpResponse]): Future[Unit] = {
-    if (condition) {
-      request.map(_ => ())
-    } else {
-      Future.successful(())
-    }
-  }
 
+      (hasAnswerChanged, needsToPayTax) match {
+        case (false, _) => Future.successful(HttpResponse(OK, ""))
+        case (true, true) => {
+          trustConnector.setTaxableTrust(request.userAnswers.identifier, needsToPayTax)
+          trustConnector.setTaxableMigrationFlag(request.userAnswers.identifier, needsToPayTax)
+        }
+        case (true, false) => trustConnector.removeTransforms(request.userAnswers.identifier)
+      }
+  }
```

Comment:
I wonder if this could perhaps return Future[Unit] and be something like:

`(hasAnswerChanged, needsToPayTax) match {
      case (false, _) => Future.successful(())
      case (true, true) =>
        for {
          _ <- trustConnector.setTaxableTrust(request.userAnswers.identifier, needsToPayTax)
          _ <- trustConnector.setTaxableMigrationFlag(request.userAnswers.identifier, needsToPayTax)
        } yield ()
      case (true, false) => trustConnector.removeTransforms(request.userAnswers.identifier).map(_ => ())
    }`

Reply:
Tidied Up